src/lib.rs

```diff
@@ -12,6 +12,7 @@
 use serde::{Deserialize, Serialize};
 
 extern crate serde;
+/// Re-export serde_json for access to error type used in the public API of this crate.
```

Comment:
In f3b23a09ec672d86e3cb7827872d1d338ba23239:

Let's just say `Re-export serde_json crate`. The reason given doesn't really make sense because the error type is already accessible. If you want to give a reason, you can say that it simplifies dependency management.

Reply:
I'm inclined to just put `#![allow(missing_docs)]` because writing _what_ the code does is pointless. However I'm now confused because `cargo clippy` isn't warning in my shell right now, only `cargo +nightly clippy` is and we don't run clippy with nightly in CI - so I don't know why I did this PR in the first place.

Reply:
So this code is silly
```rust
extern crate serde;
/// Re-export `serde_json` crate.
pub extern crate serde_json;
```

1. The comment is a _what_ comment not a _why_ comment
2. I have no idea why we pub export `serde_json` and not `serde` (from the code and just in generaly).

We could fix both of these problems by putting a comment that said why we pub export `serde_json`.

Reply:
The why is the what in this case. We re-export the crate so that it's available. (And this lint is absolutely idiotic.)

I'm fine to just `#[allow]` the lint if you'd like.

Also we really should `pub extern crate` serde as well ... it looks like `serde_json` was exported by me in a direct push to master 4 years ago, presumably because I needed `serde_json` specifically, and then we never revisited it.

Reply:
In general any crate we depend on for types that appear in the public API, should be re-exported, right?

> (This lint is absolutely idiotic, to be clear.)

We agree on this point :)

Reply:
With some exceptions, ofc. For example `bitcoin-internals` shouldn't be re-exported.

Reply:
Cool, got it.